pkg/driver/controller.go

```diff
@@ -58,16 +58,25 @@ const (
 	volumeParamsSecurityGroupIds              = "securityGroupIds"
 	volumeParamsSubnetIds                     = "subnetIds"
 	volumeParamsTags                          = "tags"
-	volumeParamsSkipFinalBackup               = "skipFinalBackup"
+	volumeParamsOptionsOnDeletion             = "optionsOnDeletion"
+	volumeParamsSkipFinalBackupOnDeletion     = "skipFinalBackupOnDeletion"
 	volumeParamsCopyTagsToSnapshots           = "copyTagsToSnapshots"
 	volumeParamsDataCompressionType           = "dataCompressionType"
-	volumeParamsNfsExports                    = "NfsExports"
+	volumeParamsNfsExports                    = "nfsExports"
 	volumeParamsParentVolumeId                = "parentVolumeId"
 	volumeParamsReadOnly                      = "readOnly"
 	volumeParamsRecordSizeKiB                 = "recordSizeKiB"
 	volumeParamsUserAndGroupQuotas            = "userAndGroupQuotas"
 )
 
+// The external-provisioner automatically configures reserved parameter keys on the CreateVolumeRequest:
+// https://kubernetes-csi.github.io/docs/external-provisioner.html#storageclass-parameters
+// Each of these parameter keys have the same prefix, "csi.storage.k8s.io".
+// We will store this prefix as a constant to avoid throwing an error when we encounter these parameters.
+const (
+	reservedVolumeParamsPrefix = "csi.storage.k8s.io"
```

Comment:
Not adding the prefix check for snapshot parameters to avoid merge conflicts. I'll address it in a future PR.